crates/ruff_python_ast/src/context.rs

```diff
@@ -300,19 +300,15 @@ impl<'a> Context<'a> {
             .map(|index| &self.scopes[*index])
     }
 
-    pub fn current_scope_grandparent(&self) -> Option<&Scope> {
+    pub fn grandparent_scope(&self) -> Option<&Scope> {
         self.scope_stack
             .iter()
             .rev()
-            .nth(2)
+            .next()
             .map(|index| &self.scopes[*index])
```

Comment:
I don't think this returns the `grandparent` scope

* `scope_stack.iter()` returns the reversed stack
* `next` returns the first element 

-> That would mean it returns the global scope. 

I believe this should instead be

```rust
self.scope_stack.iter().nth(2).map(|index| &self.scopes[*index])
```

Reply:
Yeah, it used to be like this but now `.nth(2)` returns the function scope (current scope) and `.nth(0)` or `.next()` returns the module scope. Or am I thinking wrong: `function < class < module <=> current < parent < grandparent`